modules/perception/map/hdmap/hdmap_input.cc

```diff
@@ -163,9 +163,8 @@ void HDMapInput::MergeBoundaryJunction(
   road_boundaries_ptr->resize(polygon_size);
   int polygons_index = 0;
   // merege boundary
-  int step = hdmap_sample_step_;
   PointDCloudPtr temp_cloud = base::PointDCloudPool::Instance().Get();
-  for (int i = 0; i < polygon_size; i++) {
+  for (int step = 0, i = 0; i < polygon_size; i++) {
```

Comment:
No need to reuse the same variable. Simply declare temporary ones every time.

Reply:
step is more meaningful in this situation.